tool/templates/js/runtime.mjs

```diff
@@ -214,42 +214,56 @@ export class DiplomatWriteBuf {
 }
 
 /**
- * A number of Rust functions in WebAssembly require a buffer to populate struct, slice, Option<> or Result<> types with information.
- * {@link DiplomatReceiveBuf} allocates a buffer in WebAssembly, which can then be passed into functions with the {@link DiplomatReceiveBuf.buffer}
- * property.
+ * Represents an underlying slice that we've grabbed from WebAssembly.
+ * You can treat this in JS as a regular slice of primitives, but it handles additional data for you behind the scenes.
  */
-export class DiplomatReceiveBuf {
+export class DiplomatSlice {
     #wasm;
 
-    #size;
-    #align;
-
-    #hasResult;
+    #bufferType;
+    get bufferType() {
+        return this.#bufferType;
+    }
 
     #buffer;
+    get buffer() {
+        return this.#buffer;
+    }
 
-    constructor(wasm, size, align, hasResult) {
+    #lifetimeEdges;
+
+    constructor(wasm, buffer, bufferType, lifetimeEdges) {
         this.#wasm = wasm;
+        
+        const [ptr, size] = new Uint32Array(this.#wasm.memory.buffer, buffer, 2);
 
-        this.#size = size;
-        this.#align = align;
+        this.#buffer = new bufferType(this.#wasm.memory.buffer, ptr, size);
+        this.#bufferType = bufferType;
 
-        this.#hasResult = hasResult;
+        this.#lifetimeEdges = lifetimeEdges;
+        if (this.#lifetimeEdges.length === 0) {
+            this.#wasm.diplomat_free(ptr, size, this.#bufferType.BYTES_PER_ELEMENT);
+        }
```

Comment:
Question/Issue: This `DiplomatSlice` type is here should be retaining the memory of the slice it points to. It doesn't own anything, so it shouldn't ever free it itself, right?

Reply:
This was definitely a mistake on my part. Dart moves freeing into a custom class for slices, but we already handle this. I'll retain lifetime edges if you'd like, but I'm removing the freeing code.